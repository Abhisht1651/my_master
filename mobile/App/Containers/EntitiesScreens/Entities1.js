// @flow
import { NavigationContainerProps } from '@react-navigation/native'
import React from 'react'
import { View, Text, TouchableOpacity, Image, ScrollView } from 'react-native'
import images from '../../Themes/Images'
import Bubbles2 from '../../Components/Icons/Bubbles2'
import ThumbsLarge from '../../Components/Entities/ThumbsLarge'
import Profile from '../../Components/Entities/Profile'
import { X4SButton } from '../../Components/Buttons/X4SButton'

// Styles
import styles from './ScreenStyles/Entities1Styles'

type Props = {
  title: string,
  about: string,
  count: number,
  details: string
} & NavigationContainerProps

const Entities1 = ({ title, about, count, details, navigation }: Props) => {
  return (
    <View style={styles.screen}>
      <View style={styles.topBox}>
        <TouchableOpacity onPress={() => navigation.navigate('HomeScreen')}>
          <Image source={images.arrow} />
        </TouchableOpacity>
        <Bubbles2 />
      </View>

      <View style={styles.contentBox}>
        <View style={styles.smallContentBox}>
          <View style={styles.profilepicBox}>
            <Profile profileIcons={[images.EntitiesImage11]} />
          </View>
          <View style={styles.profileDetails}>
            <Text style={styles.profileTitleFont}>{title}</Text>
            <Text style={styles.profileAboutFont}>{about}</Text>
          </View>
          <View style={styles.followContentBox}>
            <X4SButton outline title="Follow" />
            <Text style={styles.followDetails}>{`${count}K Following`}</Text>
          </View>
          <View style={styles.profileDetailsBox}>
            <Text style={styles.detailTitleFont}>Details</Text>
            <Text style={styles.detailsFont}>{details}</Text>
          </View>
        </View>
      </View>

      <View style={styles.footerBox}>
        <View style={styles.footerBox1}>
          <View style={styles.textBox}>
            <Text style={styles.textFont1}>Similar Entities</Text>
            <Text style={styles.textFont2}>View All</Text>
          </View>
        </View>
        <View style={styles.footerBox2}>
          <ScrollView horizontal>
            <View style={styles.box2}>
              <ThumbsLarge title="India" profileIcons={[images.EntitiesImage0]} />
            </View>
            <View style={styles.box}>
              <ThumbsLarge title="USA" profileIcons={[images.EntitiesImage3]} />
            </View>
            <View style={styles.box} profileIcons={[images.EntitiesImage4]}>
              <ThumbsLarge title="UNICEF" profileIcons={[images.EntitiesImage4]} />
            </View>
            <View style={styles.box}>
              <ThumbsLarge title="European" profileIcons={[images.EntitiesImage6]} />
            </View>
            <View style={styles.box}>
              <ThumbsLarge title="Europe" profileIcons={[images.EntitiesImage5]} />
            </View>
            <View style={styles.box}>
              <ThumbsLarge />
            </View>
            <View style={styles.box}>
              <ThumbsLarge />
            </View>
          </ScrollView>
        </View>
      </View>
    </View>
  )
}

Entities1.defaultProps = {
  title: 'Donald Trump',
  about: 'President of the USA',
  count: 54,
  details:
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam.'
}

export default Entities1
