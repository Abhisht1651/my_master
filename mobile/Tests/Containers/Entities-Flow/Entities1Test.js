import React from 'react'
import { shallow } from 'enzyme'
import { Image } from 'react-native'
import { TouchableOpacity, Text } from 'react-native'
import Entities1 from '../../../App/Containers/EntitiesScreens/Entities1'
// Test for Entities (snapshot and props)
describe('Entities1', () => {
  let wrapper
  beforeEach(() => {
    wrapper = shallow(<Entities1 route={{ params: { selected: [] } }} />) // no compile-time error
  })
  it('Entities1 component renders correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })
  it('should render 1 TouchableOpacity', () => {
    expect(wrapper.find(TouchableOpacity)).toHaveLength(1)
  })
  it('should render 7 Text', () => {
    expect(wrapper.find(Text)).toHaveLength(7)
  })
})
