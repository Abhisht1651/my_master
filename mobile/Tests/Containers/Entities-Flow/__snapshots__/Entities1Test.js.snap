// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Entities1 Entities1 component renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "elevation": 2,
        "flexDirection": "row",
        "height": 100.21428571428572,
        "justifyContent": "space-between",
        "paddingHorizontal": 50,
        "shadowColor": "#E6ECED",
        "width": "100%",
      }
    }
  >
    <ForwardRef
      onPress={[Function]}
    >
      <Image
        source={
          Object {
            "default": "",
          }
        }
      />
    </ForwardRef>
    <Bubbles2 />
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "elevation": 1,
        "height": 847.7142857142857,
        "justifyContent": "center",
        "paddingBottom": 95.28571428571428,
        "paddingLeft": 50,
        "paddingRight": 62,
        "paddingTop": 60.785714285714285,
        "shadowColor": "#E6ECED",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Object {
            "marginTop": 60.785714285714285,
          }
        }
      >
        <Profile
          profileIcons={
            Array [
              Object {
                "default": "",
              },
            ]
          }
        />
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "height": 80.5,
            "justifyContent": "center",
            "marginTop": 31.21428571428571,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#1F1F60",
              "fontFamily": "BananaGrotesk-Bold",
              "fontSize": 40,
              "letterSpacing": -0.28,
            }
          }
        >
          Donald Trump
        </Text>
        <Text
          style={
            Object {
              "color": "#1F1F60",
              "fontFamily": "BananaGrotesk-Light",
              "fontSize": 32,
              "letterSpacing": -0.22,
              "textAlign": "left",
            }
          }
        >
          President of the USA
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "height": 93.64285714285714,
            "justifyContent": "center",
            "marginTop": 49.28571428571429,
          }
        }
      >
        <X4SButton
          disabled={false}
          onPress={[Function]}
          outline={true}
          style={Object {}}
          title="Follow"
        />
        <Text
          style={
            Object {
              "color": "#8F8FAF",
              "fontFamily": "BananaGrotesk-Light",
              "fontSize": 28.000000000000004,
              "marginTop": 8.214285714285715,
            }
          }
        >
          54K Following
        </Text>
      </View>
      <View
        style={
          Object {
            "marginBottom": 95.28571428571428,
            "marginTop": 47.64285714285714,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#1F1F60",
              "fontFamily": "BananaGrotesk-Medium",
              "fontSize": 36,
              "letterSpacing": -0.24,
            }
          }
        >
          Details
        </Text>
        <Text
          style={
            Object {
              "color": "#1F1F60",
              "fontFamily": "BananaGrotesk-Light",
              "fontSize": 32,
              "letterSpacing": -0.22,
              "marginTop": 24.642857142857146,
            }
          }
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam.
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "height": 302.2857142857143,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "height": 138,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "flex-end",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginHorizontal": 50,
            "marginTop": 69,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#1F1F60",
              "fontFamily": "BananaGrotesk-Medium",
              "fontSize": 36,
              "letterSpacing": -0.24,
              "opacity": 1,
              "textAlign": "left",
            }
          }
        >
          Similar Entities
        </Text>
        <Text
          style={
            Object {
              "color": "#7B46E4",
              "fontFamily": "BananaGrotesk-Medium",
              "fontSize": 32,
              "letterSpacing": -0.18,
              "opacity": 1,
              "textAlign": "left",
            }
          }
        >
          View All
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "height": 179.07142857142858,
        }
      }
    >
      <ScrollViewMock
        horizontal={true}
      >
        <View
          style={
            Object {
              "paddingLeft": 26,
              "paddingRight": 8,
            }
          }
        >
          <ThumbsLarge
            profileIcons={
              Array [
                Object {
                  "default": "",
                },
              ]
            }
            title="India"
          />
        </View>
        <View
          style={
            Object {
              "paddingHorizontal": 8,
            }
          }
        >
          <ThumbsLarge
            profileIcons={
              Array [
                Object {
                  "default": "",
                },
              ]
            }
            title="USA"
          />
        </View>
        <View
          profileIcons={
            Array [
              Object {
                "default": "",
              },
            ]
          }
          style={
            Object {
              "paddingHorizontal": 8,
            }
          }
        >
          <ThumbsLarge
            profileIcons={
              Array [
                Object {
                  "default": "",
                },
              ]
            }
            title="UNICEF"
          />
        </View>
        <View
          style={
            Object {
              "paddingHorizontal": 8,
            }
          }
        >
          <ThumbsLarge
            profileIcons={
              Array [
                Object {
                  "default": "",
                },
              ]
            }
            title="European"
          />
        </View>
        <View
          style={
            Object {
              "paddingHorizontal": 8,
            }
          }
        >
          <ThumbsLarge
            profileIcons={
              Array [
                Object {
                  "default": "",
                },
              ]
            }
            title="Europe"
          />
        </View>
        <View
          style={
            Object {
              "paddingHorizontal": 8,
            }
          }
        >
          <ThumbsLarge
            profileIcons={
              Array [
                Object {
                  "default": "",
                },
              ]
            }
            title="Default"
          />
        </View>
        <View
          style={
            Object {
              "paddingHorizontal": 8,
            }
          }
        >
          <ThumbsLarge
            profileIcons={
              Array [
                Object {
                  "default": "",
                },
              ]
            }
            title="Default"
          />
        </View>
      </ScrollViewMock>
    </View>
  </View>
</View>
`;
